### DISCUSSION TITLE
sendRequst enhancements

### DISCUSSION THREAD

**Discussion by cyal1** [PortSwigger/burp-extensions-montoya-api#65] [Ideas]
Hello, can montoya api implement sending requests using other proxies, like sendRequestWithProxy or sendRequst().withHttpProxy() and the like.
https://portswigger.github.io/burp-extensions-montoya-api/javadoc/burp/api/montoya/http/Http.html#sendRequest(burp.api.montoya.http.message.requests.HttpRequest,burp.api.montoya.http.HttpMode)

**Comment by Hannah-PortSwigger** [comment]
Hi

Extensions using Burp's networking methods will respect Burp's networking configuration. This includes any upstream proxies that have been configured under "Settings > Network > Connections > Upstream proxy servers".

**Reply by cyal1** [reply]
Hello,
I want write a context menu extender, send the request of the message editor or the selected list of several requests through the proxy alone, and do not want all requests to go through the proxy, only I selected.
Do you have any good suggestions, thanks.

**Reply by Hannah-PortSwigger** [reply]
You can find an example extension that demonstrates working with context menus here: https://github.com/PortSwigger/burp-extensions-montoya-api-examples/tree/main/contextmenu

We'd recommend using the example extensions as a base to quickly get started.

**Reply by cyal1** [reply]
I am very good at creating context menus. My question is how to let a certain request I selected go through the proxy instead of all, using the sendRequest method.

**Reply by Hannah-PortSwigger** [reply]
Hi

In order to pass through the Proxy tool, the request needs to be proxied through Burp, rather than originating from Burp.

You may be able to get around this by using Java's networking library (rather than Burp's) to perform the request, with a proxy set to the same as your Proxy Listener (by default, 127.0.0.1:8080).
